x/stake/types/msg.go

```diff
@@ -321,18 +321,18 @@ func NewMsgBeginUnbonding(delegatorAddr, validatorAddr sdk.Address, sharesAmount
 }
 
 //nolint
-func (msg MsgBeginUnbonding) Type() string              { return MsgType }
-func (msg MsgBeginUnbonding) GetSigners() []sdk.Address { return []sdk.Address{msg.DelegatorAddr} }
+func (msg MsgBeginUnbonding) Type() string                 { return MsgType }
+func (msg MsgBeginUnbonding) GetSigners() []sdk.AccAddress { return []sdk.AccAddress{msg.DelegatorAddr} }
 
 // get the bytes for the message signer to sign on
 func (msg MsgBeginUnbonding) GetSignBytes() []byte {
 	b, err := MsgCdc.MarshalJSON(struct {
-		DelegatorAddr string `json:"delegator_addr"`
-		ValidatorAddr string `json:"validator_addr"`
-		SharesAmount  string `json:"shares_amount"`
+		DelegatorAddr sdk.AccAddress `json:"delegator_addr"`
```

Comment:
Can we get rid of the extra struct now?

Reply:
For this one, sharesAmount needs to be a string (not sure why though)